### PR TITLE
Addition to view settings refactor

### PR DESCRIPTION
**Pull request by FichteFoll**
Address my comments in the PR https://github.com/SublimeLinter/SublimeLinter/pull/825.

### CODE REVIEW

**Comment by kaste** [comment]
This is 🎆 . I think I put these commits on top of my PR. I will likely change the data structure of `variables` to a `ChainMap` bc this is a hot path function and ChainMap is faster here.

EDIT: Ah this is already a PR on my fork. Well done, Sir.

File: lint/linter.py [hunk 1]
```diff
@@ -444,44 +444,51 @@ def get_view_settings(self):
...
         """
-        def recursive_replace(vars, value):
+        def recursive_replace(variables, value):
             if isinstance(value, Mapping):
-                return {key: recursive_replace(vars, val)
+                return {key: recursive_replace(variables, val)
                         for key, val in value.items()}
-            elif isinstance(value, list):
-                return [recursive_replace(vars, item)
+            elif isinstance(value, Sequence):
```
Comment (kaste) [inline lint/linter.py:470]: Note 🐛 : strings match collections.Sequence

Comment (FichteFoll) [reply lint/linter.py:470]: Ah right. Should check for strings first then.

**Comment by FichteFoll** [comment]
I should use chain maps more often. I only discovered that structure a short while ago, so I guess that's why it's not always on the top of my head.

File: lint/linter.py (hunk 1 above)
Comment (kaste) [reply lint/linter.py:470]: Yup, done.

**Pull request merged**